pyinteraph/path_analysis.py

```diff
@@ -0,0 +1,481 @@
+    """Take in a string e.g. A12:A22,A13... and res names if a PDB file 
+    is supplied. Replaces the range indicated by the colon with all 
+    resiues in that range and keeps all residues separated by commas. 
+    Removes duplicates. Takes in a string e.g.1,3,4:56 if no PDB file
+    is supplied.
+    """
+
+    # Check if PDB file is supplied
+    if pdb:
+        # Find all residues sepearted by commas
+        input_comma = re.sub('\w\d+:\w\d+', '', user_input)
+        # Find all residues sepearted by xolona
+        input_colon = re.findall('\w\d+:\w\d+', user_input)
+    else:
+        # No PDB file present
+        input_comma = re.sub('\d+:\d+', '', user_input)
+        input_colon = re.findall('\d+:\d+', user_input)
+    comma_list = input_comma.split(',')
+    # Remove empty residues
+    comma_list = [res for res in comma_list if res != '']
+    # Report if any residues are not in the PDB
+    try:
+        for res in comma_list:
+            identifiers.index(res)
+    except Exception:
+        error_str = "Residue not in PDB: {:s}"
+        raise ValueError(error_str.format(res))
+    colon_replace = []
+    # Subsitite range of residues with the actual residues
+    for inp in input_colon:
+        try:
+            # Create list of size two with start and end of range
+            colon_split = inp.split(':')
+            # Find the index of those res in the indentifiers list
+            index = [identifiers.index(res) for res in colon_split]
+            # Replace with the residues in that range
+            inp_replace = identifiers[index[0]:index[1]+1]
+            # Concatenate to list
+            colon_replace += inp_replace
+        except Exception:
+            # Report if the specified range does not exist in the PDB
+            error_str = "Residue range not in PDB or incorrect format: {:s}"
+            raise ValueError(error_str.format(inp))
+    # Add both lists
+    input_list = comma_list + colon_replace
+    # Remove duplicates
+    input_list = list(set(input_list))
+    return input_list
+
+def get_shortest_paths(graph, source, target, maxl):
+    """Find all shortest paths between all combinations of source and
+    target.
+    """
+    
+    # Get all combinations
+    combinations = itertools.product(source, target)
+    # Get all shortest paths
+    paths = []
+    for node1, node2 in combinations:
+        if node1 != node2:
+            try:
+                # Get a list of shortest paths and append to output
+                path = list(nx.algorithms.shortest_paths.generic.all_shortest_paths(\
+                                G = graph, \
+                                source = node1, \
+                                target = node2))
+                for p in path:
+                    # Check that path is not longer than the maximum allowed length
+                    if len(p) <= maxl:
+                        paths.append(p)
+            except nx.NetworkXNoPath:
+                pass
```

Comment:
You do need the `pass` here since you are doing the logging in the next line (`pass` is used when the code block would be completely empty otherwise, but this is not the case).